### PR TITLE
sql: populate role valid until fields in pg_catalog tables

### LINKED ISSUE #49420: sql: populate role valid until fields in pg_catalog tables
Populate `rolvaliduntil` in pg_catalog tables.

Right now, `rolvaliduntil` is always NULL in pg_catalog.pg_roles and pg_catalog.authid.

Example: test does not have rolvaliduntil value in pg_catalog.pg_roles as '2021-01-01'.

Should be an easy fix similar to #49207.
```
root@127.0.0.1:50287/movr> CREATE USER test VALID UNTIL '2021-01-01';
sCREATE ROLE

Time: 3.72ms

root@127.0.0.1:50287/movr> SELECT * FROM pg_catalog.pg_roles;
     oid     | rolname | rolsuper | rolinherit | rolcreaterole | rolcreatedb | rolcatupdate | rolcanlogin | rolreplication | rolconnlimit | rolpassword | rolvaliduntil | rolbypassrls | rolconfig
-------------+---------+----------+------------+---------------+-------------+--------------+-------------+----------------+--------------+-------------+---------------+--------------+------------
  2310524507 | admin   |   true   |    true    |     true      |    true     |    false     |    true     |     false      |           -1 | ********    | NULL          |    false     | NULL
  1546506610 | root    |   true   |   false    |     true      |    true     |    false     |    true     |     false      |           -1 | ********    | NULL          |    false     | NULL
  4118688310 | test    |  false   |   false    |     false     |    false    |    false     |    true     |     false      |           -1 | ********    | NULL          |    false     | NULL
(3 rows)
```

### PR DESCRIPTION
**Pull request by Gorjunov**
Fixes #49420.

Previously, the `rolvaliduntil` value was always `NULL` for `pg_roles` and `pg_authid`.

Release note (bug fix): Correctly populate rolvaliduntil value for roles in pg_roles and pg_authid.

### CODE REVIEW

**Review by RichardJCai — COMMENTED** [review]
Thanks a lot for your PR! Looks great overall!

Also I used the wrong Release note tag in my PR, can you change yours to `Release note (bug fix)`.

Once again thanks so much.

File: pkg/sql/information_schema.go [hunk 1]
```diff
@@ -1817,23 +1818,18 @@ func forEachColumnInIndex(
 }
 
 func forEachRole(
-	ctx context.Context, p *planner, fn func(username string, isRole bool, noLogin bool) error,
+	ctx context.Context, p *planner, fn func(username string, isRole bool, noLogin bool, rolValidUntil time.Time) error,
 ) error {
 	query := `
```
[the remark this answers is not in the document]
Comment (Gorjunov) [reply pkg/sql/information_schema.go:1823]: Fixed

**Review by RichardJCai — COMMENTED** [review]
Well done! Looks good to me now. There is still a linter error in CI, if you run `make lintshort` it should show you what to do.

**Comment by Gorjunov** [comment]
@RichardJCai Thank you very much my first PR!
Looks it is ready now.

**Review by RichardJCai — APPROVED** [review]

**Comment by RichardJCai** [comment]
Merging it, thanks for the PR @Gorjunov !

**Comment by SWDevAngel** [comment]
Hi Sergei. Thank you for contributing to CockroachDB this year. As a token of our appreciation we would like to send you a gift. Please DM me on our [community Slack](https://cockroa.ch/slack) @lisa so I can send you a link. (If you don’t want a gift, we also have a charitable contribution choice.) All orders need to be in by December 13, so please send this asap. :)

**Comment by SWDevAngel** [comment]
@Gorjunov LAST DAY TO ORDER!
Hi Sergei. Thank you for contributing to CockroachDB this year. As a token of our appreciation we would like to send you a gift. We tried to reach you via GitHub or on email, but if you didn’t see the message please DM me on our [community Slack](https://cockroa.ch/slack) `@lisa` so I can send you a link to order your gift. (If you don’t want a gift, we also have a charitable contribution choice.) All orders must be in by December 13 (that’s tomorrow!), so please send this asap. :)

**Pull request merged**